ui/src/timeMachine/actions/queries.ts

```diff
@@ -227,6 +232,11 @@ export const executeQueries = (abortController?: AbortController) => async (
       const extern = buildVarsOption(variableAssignments)
 
       event('runQuery', {context: 'timeMachine'})
+      if (isCurrentPageDashboard(state)) {
+        // reset any existing matching query in the cache
+        resetQueryCacheByQuery(text)
```

Comment:
I don't get this- why are we resetting cache for text and then checking to see if the text is in cache?

Reply:
That's a great question. I think the use case here is:

Cell A has a query `|> foo` with no variables
Cell B has a query `|> bar` with no variables

Cell B is configured and the query is updated to `|> foo`. We want to make sure that we don't end up pulling up a cached value. In order to do that, we first need to clear out whatever value current exists in the cache (for the new query, since it might not be "new")